server/embed/config.go

```diff
@@ -211,12 +211,12 @@ type Config struct {
 	// streams that each client can open at a time.
 	MaxConcurrentStreams uint32 `json:"max-concurrent-streams"`
 
-	ListenPeerUrls, ListenClientUrls       []url.URL
-	AdvertisePeerUrls, AdvertiseClientUrls []url.URL
-	ClientTLSInfo                          transport.TLSInfo
-	ClientAutoTLS                          bool
-	PeerTLSInfo                            transport.TLSInfo
-	PeerAutoTLS                            bool
+	ListenPeerUrls, ListenClientUrls, ListenClientHttpUrls []url.URL
```

Comment:
As mentioned during today's discussions. 
Maybe to avoid changing fields semantic, we should rather be very explicit here, i.e. introduce
```
  --listen_client_http_urls  -> that servers only http handlers
  --listen_client_grpc_urls -> that servers only grpc 
```
And `listen_client_urls` stays as the cmux'ed variant of both that is discouraged...

In v3.6 we will make a call based on cmux and golang http server health whether we 
swap the defaults and default `listen_client_urls` to '', while `listen_client_grpc_urls` gets the original endpoint. 

But thanks to introduction of 2 flags: 
 a) We still have an option to be fully backward compatible in v3.6
 b) We minimize the backported change in v3.4 and v3.5.

Reply:
I have worried about dedicated `--listen-client-grpc-urls` flag. Reason is that making it work like `--listen-client-urls` is much more complicated then what is done here. Please remember that `--listen-client-urls` is used in cluster setup, like in `advertise-client-urls`. I think that would be too big change to backport. 

Just adding `--listen-client-http-urls` is much simpler as it doesn't make any changes in etcd API setup (+/- grpcgateway).